prov/util/src/util_attr.c

```diff
@@ -1122,6 +1121,37 @@ static void fi_alter_tx_attr(struct fi_tx_attr *attr,
 		attr->rma_iov_limit = hints->rma_iov_limit;
 }
 
+static uint64_t ofi_get_info_caps(const struct fi_info *prov_info,
+				  const struct fi_info *user_info,
+				  uint32_t api_version)
+{
+	int prov_mode, user_mode;
+	uint64_t caps;
+
+	caps = ofi_get_caps(prov_info->caps, user_info->caps, prov_info->caps);
+
+	prov_mode = prov_info->domain_attr->mr_mode;
+
+	if (!user_info || !ofi_rma_target_allowed(caps) ||
+	    !(prov_mode & OFI_MR_MODE_RMA_TARGET))
+		return caps;
+
+	if (!user_info->domain_attr)
+		goto trim_caps;
+
+	user_mode = user_info->domain_attr->mr_mode;
+
+	if ((FI_VERSION_LT(api_version, FI_VERSION(1,5)) &&
+	    (user_mode == FI_MR_UNSPEC)) ||
+	    (user_mode & FI_MR_BASIC) ||
```

Comment:
Is this check correct, or can we check for == BASIC?

Reply:
Yes, I think this can be reduced to "== BASIC" since user is supposed to exclude other bits when specifying BASIC.